arch/arm/core/cortex_m/tz/arm_core_tz.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arch/arm/cortex_m/cmsis.h>
+#include <cortex_m/tz.h>
+
+static void configure_nonsecure_vtor_offset(u32_t vtor_ns)
+{
+	SCB_NS->VTOR = vtor_ns;
+}
+
+static void configure_nonsecure_msp(u32_t msp_ns)
+{
+	__TZ_set_MSP_NS(msp_ns);
+}
+
+static void configure_nonsecure_psp(u32_t psp_ns)
+{
+	__TZ_set_PSP_NS(psp_ns);
+}
+
+static void configure_nonsecure_control(u32_t spsel_ns, u32_t npriv_ns)
```

Comment:
I don't see how this could improve readability/ease-of-use for this function.
CONTROL register, clearly, has 2 bits widely used (SPSEL and nPRIV), so by having them as arguments, makes function easy to use, IMO.

Reply:
This will leave us with 2 boolean arguments, which are not so easy to read.
https://github.com/zephyrproject-rtos/zephyr/pull/6748#pullrequestreview-129070060

Reply:
Isn't this a private function? User should not be using it, AFAIK.

Reply:
Btw, the user is given a struct (properties) API to configure all Non-Secure instances of core registers.